### PR TITLE
Added `useCurrentPostId` hook

### PR DESCRIPTION
**Pull request by renatonascalves**
- As titled;
- Tweaks to the READM.md to fix some violations;
- Bumping version.

It's not clear where the source of the changelog is. So I updated in both places just in case. Both options don't seem to be respected, considering the lack of notes there. ¯\_(ツ)_/¯

### Summary by CodeRabbit

- New Feature: Introduced a new tool, `useCurrentPostId`, to the `@alleyinteractive/block-editor-tools` package. This tool allows developers to easily retrieve the ID of the current post being edited in a WordPress environment, enhancing the customization and functionality of their components.
- Documentation: Updated the package documentation and code examples to reflect the new feature and made minor textual modifications for clarity. Removed the table of contents and Slack channel support for Alley employees.
- Style: Made a minor change to the pull request message format for consistency.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -83,7 +83,7 @@ When you're finished with the changes, create a pull request, also known as a PR
   [git tutorial](https://github.com/skills/resolve-merge-conflicts) to help you
   resolve merge conflicts and other issues.
 
-#### Your PR is Merged!
+#### Your PR is Merged
```
Comment (renatonascalves) [inline CONTRIBUTING.md:86]: See https://github.com/DavidAnson/markdownlint/blob/v0.31.1/doc/md026.md

File: packages/block-editor-tools/README.md [hunk 2]
```diff
@@ -10,8 +10,8 @@ This package contains a set of modules used by [Alley Interactive](https://alley
 - [Usage](#usage)
 - [Changelog](#changelog)
 - [Development Process](#development-process)
-    - [Contributing](#contributing)
-	- [Releases](#Releases)
+  - [Contributing](#contributing)
+- [Releases](#releases)
```
Comment (kevinfodness) [inline packages/block-editor-tools/README.md:14]: Releases is at the same level as contributing, under development processes. I'd actually recommend just removing this entirely, as GitHub now has a table of contents for markdown files with headings bulit-in, and this takes up space and is redundant. I'll create a ticket to update our boilerplate to remove the TOC from our standard readmes.

File: CONTRIBUTING.md (hunk 1 above)
Comment (kevinfodness) [reply CONTRIBUTING.md:86]: This change kills the enthusiasm we have for merging PRs. We haven't adopted markdownlint as a standard, and I would want to review it and its ruleset before applying its recommendations.

File: packages/block-editor-tools/README.md [hunk 3]
```diff
@@ -29,25 +29,29 @@ This package assumes your project is running in an environment compatible with t
...
-	const { my_meta_key: myMetaKey } = meta;
-
-	return (
-		<TextControl
-			label={__('My Meta Key', 'alley-scripts')}
-			onChange={(newValue) => setMeta({ ...meta, my_meta_key: newValue })}
-			value={myMetaKey}
-		/>
-	);
+ const [meta, setMeta] = usePostMeta();
```
Comment (kevinfodness) [inline packages/block-editor-tools/README.md:36]: If you're going to switch tabs for spaces, that's fine, but please use 2 spaces, as that's our ESLint standard, and it's more readable that way.

File: packages/block-editor-tools/README.md [hunk 4]
```diff
@@ -72,8 +76,8 @@ This package is developed as part of the [Alley Scripts](https://github.com/alle
 
 You can contribute to this project in several ways:
 
-* Visit the main [Alley Scripts GitHub repo](https://github.com/alleyinteractive/alley-scripts) to [Open an issue](https://github.com/alleyinteractive/alley-scripts/issues/new) or submit PRs.
-* Alley employees can ask questions or ask for support in the [#javascript](https://alleyinteractive.slack.com/archives/C035Y7Q3X) channel in Slack.
+- Visit the main [Alley Scripts GitHub repo](https://github.com/alleyinteractive/alley-scripts) to [Open an issue](https://github.com/alleyinteractive/alley-scripts/issues/new) or submit PRs.
+- Alley employees can ask questions or ask for support in the [#javascript](https://alleyinteractive.slack.com/archives/C035Y7Q3X) channel in Slack.
```
Comment (kevinfodness) [inline packages/block-editor-tools/README.md:80]: I realize you're updating this and it was already there, but please unlink `#javascript` here. I don't want a link to Alley's Slack in the README for a public repo.

**Review by kevinfodness — CHANGES_REQUESTED** [review]

**Comment by renatonascalves** [comment]
CR suggestions applied. Requested another review. :)

**Review by kevinfodness — APPROVED** [review]
🍣  Thanks for making the updates!

**Pull request merged**